pkg/controller/master/upgrade/job_controller.go

```diff
@@ -29,27 +49,141 @@ type jobHandler struct {
 	planCache     upgradev1.PlanCache
 	upgradeClient ctlharvesterv1.UpgradeClient
 	upgradeCache  ctlharvesterv1.UpgradeCache
+
+	machineClient clusterv1ctl.MachineClient
+	machineCache  clusterv1ctl.MachineCache
+	nodeClient    ctlcorev1.NodeClient
+	nodeCache     ctlcorev1.NodeCache
 }
 
 func (h *jobHandler) OnChanged(key string, job *batchv1.Job) (*batchv1.Job, error) {
-	if job == nil || job.DeletionTimestamp != nil || job.Labels == nil || job.Namespace != upgradeNamespace {
+	if job == nil || job.DeletionTimestamp != nil || job.Labels == nil || (job.Namespace != upgradeNamespace && job.Namespace != sucNamespace) {
 		return job, nil
 	}
 
-	chartName := job.Labels[helmChartLabel]
+	upgradeControllerLock.Lock()
+	defer upgradeControllerLock.Unlock()
+
 	planName := job.Labels[upgradePlanLabel]
 	nodeName := job.Labels[upgradeNodeLabel]
-	if chartName == harvesterChartname {
-		return h.syncHelmChartJob(job)
-	} else if planName != "" && nodeName != "" {
-		return h.syncNodeJob(job, planName, nodeName)
+
+	switch {
+	case planName != "" && nodeName != "":
+		return h.syncPlanJob(job, planName, nodeName)
+	case job.Labels[harvesterUpgradeComponentLabel] == nodeComponent:
+		return h.syncNodeJob(job)
+	case job.Labels[harvesterUpgradeComponentLabel] == manifestComponent:
+		return h.syncManifestJob(job)
 	}
 
 	return job, nil
 }
 
-func (h *jobHandler) syncNodeJob(job *batchv1.Job, planName string, nodeName string) (*batchv1.Job, error) {
-	plan, err := h.planCache.Get(upgradeNamespace, planName)
+func (h *jobHandler) syncNodeJob(job *batchv1.Job) (*batchv1.Job, error) {
+	jobType, ok := job.Labels[upgradeJobTypeLabel]
+	if !ok {
+		return nil, errors.New("Sync a job without type")
+	}
+
+	nodeName, ok := job.Labels[harvesterNodeLabel]
+	if !ok {
+		return job, nil
+	}
+
+	node, err := h.nodeCache.Get(nodeName)
+	if err != nil {
+		return job, nil
+	}
+
+	machineName, ok := node.Annotations["cluster.x-k8s.io/machine"]
```

Comment:
I guess this should also extract as a const since both node and job controllers use it.

Reply:
Thanks. Fixed in 1b5ea72